worlds/cv64/__init__.py

```diff
@@ -0,0 +1,330 @@
+            # Empty breakables
+            if self.options.empty_breakables.value:
+                offset_data.update(rom_empty_breakables_flags)
+            # Music
+            if self.options.background_music.value:
+                offset_data.update(randomize_music(self, self.options))
+            # Loading zones
+            offset_data.update(get_loading_zone_bytes(self.options, self.starting_stage, self.active_stage_exits))
+            # Countdown
+            if self.options.countdown.value:
+                offset_data.update(get_countdown_numbers(self.options, active_locations))
+            # Start Inventory
+            offset_data.update(get_start_inventory_data(self.options, self.options.start_inventory.value, self.options.start_inventory_from_pool.value))
+
+            cv64_rom = LocalRom(get_base_rom_path())
+
+            slot_name = self.multiworld.player_name[self.player].encode("utf-8")
+
+            rompath = os.path.join(output_directory, f"{self.multiworld.get_out_file_name_base(self.player)}.z64")
+
+            patch_rom(self.multiworld, self.options, cv64_rom, self.player, offset_data, self.active_stage_exits,
+                      self.s1s_per_warp, self.active_warp_list, self.required_s2s, self.total_s2s, shop_name_list,
+                      shop_desc_list, shop_colors_list, slot_name, active_locations)
+
+            cv64_rom.write_to_file(rompath)
+
+            patch = CV64DeltaPatch(os.path.splitext(rompath)[0] + CV64DeltaPatch.patch_file_ending, player=self.player,
+                                   player_name=self.multiworld.player_name[self.player], patched_path=rompath)
+            patch.write()
+        except:
+            print(f"D'oh, something went wrong in CV64's generate_output for player "
+                  f"{self.multiworld.player_name[self.player]}!")
+            raise
+        finally:
+            if os.path.exists(rompath):
+                os.unlink(rompath)
+            self.rom_name_available_event.set()  # make sure threading continues and errors are collected
+
+    def write_spoiler(self, spoiler_handle: typing.TextIO) -> None:
+        # Write the stage order to the spoiler log
+        spoiler_handle.write(f"\nCastlevania 64 stage & warp orders for {self.multiworld.player_name[self.player]}:\n")
+        for stage in self.active_stage_list:
+            num = str(self.active_stage_exits[stage]["position"]).zfill(2)
+            path = self.active_stage_exits[stage]["path"]
+            spoiler_handle.writelines(f"Stage {num}{path}:\t{stage}\n")
+
+        # Write the warp order to the spoiler log
+        spoiler_handle.writelines(f"\nStart :\t{self.active_stage_list[0]}\n")
+        for i in range(1, len(self.active_warp_list)):
+            spoiler_handle.writelines(f"Warp {i}:\t{self.active_warp_list[i]}\n")
+
+    def fill_slot_data(self) -> typing.Dict[str, typing.Any]:
+        return {"death_link": self.options.death_link.value}
```

Comment:
Funny you mention this right when I also added the shopsanity option to it for the sake of knowing when the client should run the new shop location scouting or not for the sake of auto-hinting its things like HK does. Tho I could probably detect that in the ROM too.

Reply:
I can't reasonably demand you change it while committing the same "crime", and in the grand scheme of things it's pretty insignificant. But it does feel more proper, since you're already creating an output file.

Reply:
It is now set up to read specific bits in the ROM to detect DeathLink and shopsanity, and fill_slot_data has been done away with entirely.